background/contentScript.js

```diff
@@ -0,0 +1,12 @@
+window.addEventListener('message', function(event) {
+  // We only accept messages from ourselves
+  if (event.source !== window)
+    return;
+
+  if (event.data.from && (event.data.from === 'page')) {
+    chrome.runtime.sendMessage({
+      from: 'content',
+      data: event.data,
+    });
+  }
+}, false);
```

Comment:
A function that runs in the background of the extension and detects `postMessage`.